src/tensor_ops/binary_map.rs

```diff
@@ -158,6 +158,61 @@ where
     })
 }
 
+/// Apply binary function `f` to `lhs` and `rhs`, where `rhs` is broadcasted `M` times to be the same shape as `lhs`.
+/// `dfdx` and `dfdy` are the partial derivatives of f wrt. x and y respectively.
+///
+/// `f`, `dfdx`, and `dfdy` are all the same type.
+///
+/// Generics:
+/// - `N`: The first dimension of `lhs`.
+/// - `M`: The second dimension of `lhs`.
```

Comment:
nit: Can you make `M` the first dimension? Tried to keep `M` as the outer dimension everywhere!